spec/models/reservation_spec.rb

```diff
@@ -308,20 +311,30 @@
   end
 
   describe "#canceled?" do
+    # TODO: move to order_detail_spec
```

Comment:
Um.... `has_canceled_at`? That's basically what this is checking for, as distinct from whatever the state machine says.

Precisely what the difference in meaning is between the presence of `canceled_at` and the order_detail state of canceled is less clear to me--I still find it very weird that these are independent events. The `canceled` state seems kind of not-very-meaningful, since canceled order details can also be in `complete` state.